crates/motionfile/src/motion_interpolator.rs

```diff
@@ -51,27 +56,54 @@ impl Default for State {
 
 impl<T: Debug + Interpolate<f32>> MotionInterpolator<T> {
     pub fn advance_by(&mut self, time_step: Duration, condition_input: &ConditionInput) {
+        match self.current_state {
+            State::CheckEntry {
+                current_frame_index,
+                ..
+            }
+            | State::InterpolateSpline {
+                current_frame_index,
+                ..
+            }
+            | State::CheckExit {
+                current_frame_index,
+                ..
+            } => {
+                let current_frame = &self.frames[current_frame_index];
+                if let Some(continuous_conditions) = current_frame.enable.as_ref() {
+                    if continuous_conditions.iter().any(|condition| {
+                        matches!(condition.evaluate(condition_input), Response::Abort)
+                    }) {
+                        self.abort_motion();
+                        return;
+                    }
+                }
+            }
+            _ => (),
+        }
+
         self.current_state = match self.current_state {
             State::CheckEntry {
                 current_frame_index,
                 time_since_start,
             } => {
                 let current_frame = &self.frames[current_frame_index];
-                if current_frame
-                    .entry_condition
-                    .as_ref()
-                    .map(|condition| condition.is_fulfilled(condition_input, time_since_start))
-                    .unwrap_or(true)
-                {
-                    State::InterpolateSpline {
-                        current_frame_index,
-                        time_since_start: Duration::ZERO,
-                    }
-                } else {
-                    State::CheckEntry {
+                match current_frame.entry_condition.as_ref().map(|condition| {
+                    condition
+                        .evaluate(condition_input)
+                        .with_timeout(condition.timeout(time_since_start))
+                }) {
+                    Some(Response::Abort) => State::Aborted {
+                        at_position: self.value(),
+                    },
+                    Some(Response::Wait) => State::CheckEntry {
                         current_frame_index,
                         time_since_start: time_since_start + time_step,
```

Comment:
Why are you advancing the time in the wait state? Maybe handle it similar to below?

Reply:
I don't know what you mean with "below". The time advancement is required for the timeout functionality of conditions